controllers/gateway_policy_discoverability_reconciler.go

```diff
@@ -14,5 +28,103 @@ func NewGatewayPolicyDiscoverabilityReconciler(client *dynamic.DynamicClient) *G
 }
 
 func (r *GatewayPolicyDiscoverabilityReconciler) Subscription() *controller.Subscription {
-	return &controller.Subscription{}
+	return &controller.Subscription{
+		Events: []controller.ResourceEventMatcher{
+			{Kind: &machinery.GatewayGroupKind},
+			{Kind: &kuadrantv1beta3.AuthPolicyGroupKind},
+			{Kind: &kuadrantv1beta3.RateLimitPolicyGroupKind},
+			{Kind: &kuadrantv1alpha1.TLSPolicyGroupKind},
+			{Kind: &kuadrantv1alpha1.DNSPolicyGroupKind},
+		},
+		ReconcileFunc: r.reconcile,
+	}
+}
+
+func (r *GatewayPolicyDiscoverabilityReconciler) reconcile(ctx context.Context, _ []controller.ResourceEvent, topology *machinery.Topology, _ error, _ *sync.Map) error {
+	logger := controller.LoggerFromContext(ctx).WithName("GatewayPolicyDiscoverabilityReconciler").WithName("reconcile")
+
+	gateways := lo.FilterMap(topology.Targetables().Items(), func(item machinery.Targetable, index int) (*machinery.Gateway, bool) {
+		g, ok := item.(*machinery.Gateway)
+		return g, ok
+	})
+
+	policyKinds := []*schema.GroupKind{
+		&kuadrantv1beta3.AuthPolicyGroupKind,
+		&kuadrantv1beta3.RateLimitPolicyGroupKind,
+		&kuadrantv1alpha1.TLSPolicyGroupKind,
+		&kuadrantv1alpha1.DNSPolicyGroupKind,
+	}
+
+	for _, gw := range gateways {
+		conditions := gw.Status.DeepCopy().Conditions
+
+		// TODO: What happens for multiple policies of the same kind -
+		// TODO: Should it be conditions per listener?
```

Comment:
I would say so. TLSPolicy and DNSPolicy still don't support targeting individual listeners (https://github.com/Kuadrant/kuadrant-operator/issues/891), but the AuthPolicy and RateLimitPolicy do.

Reply:
Yeah, I've something working where the `Gateway.Status.Conditions[XPolicyAffected]`  lists xPolicies that just specifically targets the gateway. The listener status in `Gateway.Status.ListenersStatus` will list inherited gateway policies _and_ policies targetting specifically the listener. I think this is nicer and more informative to at least know which policies affect which sections of the gateway 🤔 

Example:
```yaml
conditions:
  - lastTransitionTime: "2024-10-29T09:30:44Z"
    message: Resource accepted
    observedGeneration: 1
    reason: Accepted
    status: "True"
    type: Accepted
  - lastTransitionTime: "2024-10-29T09:30:49Z"
    message: Resource programmed, assigned to service(s) kuadrant-ingressgateway-istio.gateway-system.svc.cluster.local:80
    observedGeneration: 1
    reason: Programmed
    status: "True"
    type: Programmed
  - lastTransitionTime: "2024-10-29T09:37:57Z"
    message: Object affected by RateLimitPolicy [gateway-system/gw-rlp]
    observedGeneration: 1
    reason: Accepted
    status: "True"
    type: kuadrant.io/RateLimitPolicyAffected
listeners:
  - attachedRoutes: 0
    conditions:
      - lastTransitionTime: "2024-10-29T09:30:44Z"
        message: No errors found
        observedGeneration: 1
        reason: Accepted
        status: "True"
        type: Accepted
      - lastTransitionTime: "2024-10-29T09:30:44Z"
        message: No errors found
        observedGeneration: 1
        reason: NoConflicts
        status: "False"
        type: Conflicted
      - lastTransitionTime: "2024-10-29T09:30:44Z"
        message: No errors found
        observedGeneration: 1
        reason: Programmed
        status: "True"
        type: Programmed
      - lastTransitionTime: "2024-10-29T09:30:44Z"
        message: No errors found
        observedGeneration: 1
        reason: ResolvedRefs
        status: "True"
        type: ResolvedRefs
      - lastTransitionTime: "2024-10-29T09:37:39Z"
        message: Object affected by RateLimitPolicy [gateway-system/gw-rlp gateway-system/section-rlp]
        observedGeneration: 1
        reason: Accepted
        status: "True"
        type: kuadrant.io/RateLimitPolicyAffected
    name: http
    supportedKinds:
      - group: gateway.networking.k8s.io
        kind: HTTPRoute
      - group: gateway.networking.k8s.io
        kind: GRPCRoute
```

What do you think? 🤔